lib/stdlib/test/property_test/queue_prop.erl

```diff
@@ -685,51 +680,45 @@ do_op_list(Op, [_|L1])
        Op =:= tail ->
     {undefined, L1}.
 
-%% in/2, in_r/2 and cons/2 insert the item into the queue
-do_op_queue({Op, I}, Q0)
-  when Op =:= in;
-       Op =:= in_r;
-       Op =:= cons ->
-    {undefined, queue:Op(I, Q0)};
-%% snoc/2 inserts the item into the queue
-do_op_queue({snoc, I}, Q0) ->
-    {undefined, queue:snoc(Q0, I)};
-%% out/1 and out_r/1 remove and return an item from the queue
-do_op_queue(Op, Q0)
-  when Op =:= out;
-       Op =:= out_r ->
-    queue:Op(Q0);
-%% peek/1 and peek_r/1 return but do not remove an item from the queue
-do_op_queue(Op, Q0)
-  when Op =:= peek;
-       Op =:= peek_r ->
-    {queue:Op(Q0), Q0};
-%% get/1, get_r/1, head/1, last/1 and daeh/1 return but do not remove
-%% an item from the queue
-do_op_queue(Op, Q0)
-  when Op =:= get; Op =:= get_r;
-       Op =:= head; Op =:= daeh;
-       Op =:= last ->
-    try
-        queue:Op(Q0)
-    of
-        I -> {{value, I}, Q0}
-    catch
-        error:empty -> {empty, Q0}
-    end;
-%% init/1, drop/1, drop_r/1, tail/1 and liat/1 remove an item from the
-%% queue
-do_op_queue(Op, Q0)
-  when Op =:= init;
-       Op =:= drop; Op =:= drop_r;
-       Op =:= tail; Op =:= liat ->
-    try
-        queue:Op(Q0)
-    of
-        Q1 -> {undefined, Q1}
-    catch
-        error:empty -> {undefined, Q0}
-    end.
+queue_op_map() ->
+    SwapArgs = fun (F) -> fun (X, Y) -> F(Y, X) end end,
+    QueueOp1 = fun (F) -> fun (Q) -> {undefined, F(Q)} end end,
+    QueueOp2 = fun (F) -> fun (I, Q) -> {undefined, F(I, Q)} end end,
+    WrappedReturn = fun (F) -> fun (Q) -> {F(Q), Q} end end,
+    PlainReturn = fun (F) -> fun (Q) -> {{value, F(Q)}, Q} end end,
+    CatchEmpty = fun (F) ->
+        fun (Q) ->
+            try F(Q) catch error:empty -> {empty, Q} end
```

Comment:
This and the above is much better after changing the semantics of `do_op_list/2`.  Much better names than I propposed as well.
Minor point on formatting: in the OTP code base, about 90% of the existing anonymous are written as `fun(..) -> .. `, i.e., with no space after `fun`.

Reply:
Ok, I changed that in my last commit.
Personally, I prefer a more condensed style with fewer spaces. When I'm working in other code bases that use a more spacey style, I seem to over-adapt and sprinkle spaces all over the place :sweat_smile: